pyintegration/deephaven2/agg.py

```diff
@@ -16,257 +20,288 @@ class Aggregation:
     Note: It should not be instantiated directly by user code but rather through the static methods on the class.
     """
 
-    def __init__(self, j_aggregation):
-        self._j_aggregation = j_aggregation
+    def __init__(self, j_agg_spec, cols: List[str] = None):
+        self._j_agg_spec = j_agg_spec
+        self._cols = cols
 
     @property
-    def j_agg(self):
-        return self._j_aggregation
+    def j_aggregation(self):
+        if self._cols:
+            if not self._j_agg_spec:
+                # special case for count()
+                return _JAggregation.AggCount(self._cols[0])
+
+            return self._j_agg_spec.aggregation(*[_JPair.parse(col) for col in self._cols])
+        else:
+            raise DHError("No columns specified for the aggregation operation.")
 
+    @property
+    def j_agg_spec(self):
+        return self._j_agg_spec
 
-def sum_(cols: List[str]) -> Aggregation:
+
+def sum_(cols: List[str] = None) -> Aggregation:
     """ Create a Sum aggregation.
 
     Args:
-        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"
+        cols (List[str]): the columns to aggregate on, can be renaming expressions, i.e. "new_col = col"; default is
+            None, only valid when used in Table agg_all_by operation
```

Comment:
Note that the docstring and default highlight the awkwardness of how the column names are handled.  We are defaulting to value for a method that is rarely used.

Reply:
.... I finally reviewed everything.  I'm not opposed to the change, but I am leaving the comments in case they trigger some thoughts.